pkg/providers/v1/aws.go

```diff
@@ -1668,7 +1668,16 @@ func (c *Cloud) NodeAddressesByProviderID(ctx context.Context, providerID string
 		if eni == nil || err != nil {
 			return nil, err
 		}
-		return getNodeAddressesForFargateNode(aws.StringValue(eni.PrivateDnsName), aws.StringValue(eni.PrivateIpAddress)), nil
+
+		// Assign NodeInternalIP based on IP family
+		if eni.Ipv6Addresses != nil {
```

Comment:
Does this mean node IPs are either/or? I think this needs to respect `c.cfg.Global.NodeIPFamilies`

Reply:
Updated the PR to set the IP based on NodeIPFamilies consistent with EC2 nodes and verified that the addresses field with cloud provider is consistent with addresses field without cloud provider enabled